### PR TITLE
Added user privilege level

### PR DESCRIPTION
**Pull request by woogerboy21**
Added a enum column in the users table named "privilevel" with the current values of "none", "vip", and "donator".  Also allowed anyone with a higher privilege level than "none" to log in even if the server
is set to limit the user total and the user limit is reached.  This change add's the new user information into the users container that gets populated and passed between client and server.

### CODE REVIEW

**Comment by ZeldaZach** [comment]
Would you like to explain what VIP and DONOR mean exactly? Just making sure we have the right number and types of enums

**Comment by woogerboy21** [comment]
The idea of the enum was that there could be additional levels added later as some have expressed interest in having them.  But the defaults that I added was VIP (for admin/moderators) and then Donators would be anyone that has provided time to either the project or has given a donation of some type to the servers owner / operator.

**Comment by woogerboy21** [comment]
Should this be seen as a "feature"?  It is something both the client and server recognize, though the server is the only thing so far that does anything with the data really.

**Comment by woogerboy21** [comment]
Waiting for @Daenyth to take a look (since he expressed interest in doing so).  Though the change is pretty basic.

**Comment by woogerboy21** [comment]
FYI, I still hate git .....

File: servatrice/migrations/servatrice_0017_to_0018.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- Servatrice db migration from version 17 to version 18
+
+alter table cockatrice_users add privlevel enum("NONE","VIP","DONATOR") NOT NULL;
```
Comment (Daenyth) [inline servatrice/migrations/servatrice_0017_to_0018.sql]: Shouldn't this be `ADD COLUMN`, not `ADD`?

File: servatrice/servatrice.ini.example [hunk 2]
```diff
@@ -39,6 +39,8 @@ writelog=1
 
 ; Choose a name for the log file, if enabled; you can specify an absolute path or a path relative to
 ; the servatrice executable; the default file name is server.log (in the same path as servatrice)
+; Note: When running servatrice under windows you will need to use double backslashes between folder locations
```
Comment (Daenyth) [inline servatrice/servatrice.ini.example]: 👍

File: servatrice/migrations/servatrice_0017_to_0018.sql (hunk 1 above)
Comment (woogerboy21) [inline servatrice/migrations/servatrice_0017_to_0018.sql]: This actually works both ways (with or without the word column).  At least in the version of mariadb that I am running.  Would you like me to add the word column to the script?

**Pull request merged**